### DISCUSSION TITLE
Components with client:only directive in MDX?

### DISCUSSION THREAD

**Discussion by rsimon** [withastro/starlight#2235] [Q&A]
I want to add an interactive component which needs to run in the browser instead of pre-rendering. In an Astro page, I can do this by setting the [client:only](https://docs.astro.build/en/reference/directives-reference/#clientonly) directive. Is there a way I can use this directive in MDX pages, too?

**Comment by delucis** [comment]
I think it should work in MDX too? Did you try?

**Comment by rsimon** [comment]
Yes :-) I got a `window not defined` error. FWIW: it's not a big deal if client:only components don't work in MDX. For now, I resorted to `.astro` pages for dynamic content. But I thought I'd ask, to be sure I'm not missing anything.

**Reply by delucis** [reply]
Ah, interesting. I'd expect it to work, so if you can share a reproduction and open an issue in https://github.com/withastro/astro/issues, that would be great!

**Reply by rsimon** [reply]
Ok, so this _does_ seem to work after all. (I just made a minimal example, where a component with `client:only` accesses the `window` object.) However I have a library dependency that seems to trigger the `window not defined` error already when importing it. I think I can see where that might be going wrong. Looks like the library needs a small patch to make it SSR-compatible. I'll report back.

**Comment by HiDeoo** [comment]
Considering the last messages hinting this was a library issue and the lack of recent activity, I'm going to close this discussion for now.
